### DISCUSSION TITLE
Alpha stability for simple use cases

### DISCUSSION THREAD

**Discussion by gary-menzel** [KevinVandy/mantine-react-table#271] [Q&A] [answered]
While I realise an alpha may end up with breaking changes, I can see there is active development taking place.  What I would like to know is if, for very basic use, is the Alpha stable enough to work with?

By "basic use" I mean fixed column position (no dragging around), the ability to use the standard renders and such from react-table, and most importantly, that the virtualisation is stable.  It's for a "production" project but there is really only one user involved and the release would be considered "beta" for that user (i.e. also still under development).

I have upgraded to Mantine 7 but would like to start using this package.

I'm a long term user of react-table (and was an original contributor back in the V6 days) so I could potentially contribute fixes if I find anything significant (although my time is limited these days compare to before).

**Comment by gary-menzel** [comment]
Also - is there a dependency on a specific version of @tanstack/react-table ?

**Reply by KevinVandy** [reply, marked as answer]
TanStack Table is installed as a dependency instead of peer dependency internally in order to avoid differing versions requirements. So you don't have to install react table in your own package.json if you don't want to.

Since I'm also a maintainer for TanStack Table itself, I do a good job of keeping MRT up to date, most of the time shipping new TanStack versions in MRT right away.

If you need to import vanilla TanStack Table and you're using pnpm (which requires all imported packages to be explicitly installed), then it is best to make sure the same react table versions are installed to avoid duped packages. I know that can be annoying. Other package managers will let you import from TanStack Table without requiring you to list it in your package.json.

If you only plan on using MRT and not vanilla TanStack Table, don't worry about installing it at all.

**Comment by KevinVandy** [comment]
Hello fellow react-table contributor.

I've been saying that I want to move this from alpha to beta for the past couple of weeks, but have still been finalizing how all of the CSS variables work internally. That is really the main thing holding it back, but I've just about got that done.

There may still be small "breaking changes" over the next couple weeks, but I'll also be creating the official changelog and migration guide with easy to follow instructions.

As far as the normal table logic for features like virtualization, most of that code has been continuously copied over from [Material React Table](https://www.material-react-table.com/), which is stable and more popular than this lib. So I overall have some confidence the stability of most of the features.

**Reply by mamlzy** [reply]
sounds cool! really appreciate your work @KevinVandy, can't wait to see this lib getting stable!

**Comment by KevinVandy** [comment]
Just released v2.0.0-beta.0. I expect much fewer breaking changes, but bugs may still be present. The next few weeks will be docs writing and bug fixing for me.

**Comment by gary-menzel** [comment]
Hi @KevinVandy - How are things going with V2 ?

I just thought I would check in on this.  I note the current release saying Mantine V6.  I have recently upgraded to V7 (actually on 7.5.1) and wonder if there will be any major issues with V2 when it arrives or will it peer to V7 anyway?

I currently have a bare-bones wrapper of Mantine Table with TanStack React-Virtual that is just meeting my needs for now - but I do want more advanced features for the future.  Just looking at my timing for converting over to a stable version of Mantine React Table.
